pkg/apis/core/validation/shoot_test.go

```diff
@@ -4903,6 +4904,17 @@ var _ = Describe("Shoot Validation Tests", func() {
 
 				Expect(ValidateShootUpdate(newShoot, shoot)).To(BeEmpty())
 			})
+
+			It("should not sync cloud profile fields for shoots being deleted", func() {
+				shoot.Spec.CloudProfileName = ptr.To("profile")
+				shoot.DeletionTimestamp = ptr.To(metav1.Now())
+				newShoot := prepareShootForUpdate(shoot)
+
+				utils.SyncCloudProfileFields(shoot, newShoot)
+
+				errorList := ValidateShootUpdate(newShoot, shoot)
+				Expect(errorList).To(BeEmpty())
+			})
```

Comment:
Minor suggestion, you can check the newShoot and assert that there was no mutation,

Reply:
Thank you for your suggestion. I adapted (and moved) the test to be more general and to checking that `PrepareForUpdate()` does not mutate the shoot spec of the shoot being deleted. Please check again.